src/client/content_cao.cpp

```diff
@@ -1457,13 +1481,19 @@ void GenericCAO::updateAttachments()
 	}
 }
 
-void GenericCAO::readAOMessageProperties(std::istream &is)
+bool GenericCAO::visualExpiryRequired(const ObjectProperties &new_) const
 {
-	// Reset object properties first
-	m_prop = ObjectProperties();
-
-	// Then read the whole new stream
-	m_prop.deSerialize(is);
+	const ObjectProperties &old = m_prop;
+	// Ordered to compare primitive types before vectors
+	return old.is_visible != new_.is_visible ||
+		old.mesh != new_.mesh ||
+		old.spritediv != new_.spritediv ||
+		old.use_texture_alpha != new_.use_texture_alpha ||
+		old.visual != new_.visual ||
+		old.wield_item != new_.wield_item ||
+		old.visual_size != new_.visual_size ||
+		old.colors != new_.colors ||
+		old.textures != new_.textures;
```

Comment:
Following props also need to expire the visuals (in order to call `updateTextures` in `addToScene`):
`initial_sprite_basepos`
`backface_culling`